### ISSUE TITLE
Vulcanized File does not load upon bundling 

### ISSUE THREAD

**Issue by CxRes** [Hypercubed/systemjs-plugin-html#7]
Lets start with a file `bootstrap.js` that in its first line I import Polymer (I have done all the relevant installations correctly) `import 'polymer/polymer.html!'`. If I look for the function `Polymer` I find it. So far it works as expected.

Now, I run the SystemJS bundler `builder.bundle('bootstrap.js',...)`. The `html-builder.js` script creates a file `bootstrap.html` but no where is this file called in  `bootstrap.js`. There is an invocation in `bootstrap.js` of polymer:

```
System.register("bower:polymer@1.2.3/polymer.html!github:Hypercubed/systemjs-plugin-html@0.0.7", [], function() { return { setters: [], execute: function() {} } });
```

But there is no reference to `bootstrap.html` anywhere in the file and the Polymer library is not loaded. Shouldn't there be a call of the form `System.register('bootstrap.html!github:Hypercubed/systemjs-plugin-html@0.0.7,...' ? Or am i missing something?

**Comment by Hypercubed** [comment]
You need to add the `bootstrap.html` as a html import similar to how you include `bootstrap.js` and, perhaps, `boostrap.css` (see here: https://github.com/Hypercubed/Project-Chi/blob/master/app/index.html#L17).  Much like other SystemJS plugins the plugin itself is never bundled so I don't think a `System.register` will work like you have described. Let me know if this doesn't work of if you know a better way.  Thanks.

**Comment by CxRes** [comment]
@Hypercubed Honestly, I find the documentation on the plugins not clear enough but from what I gather @guybedford is holding of proper documentation until JSPM 0.19. I think though that you are right about how plugins work but it seems counter-intuitive that there is not even an option to include the bundles.

One reason I believe this is that there is an option to set `exports.build = false` in the plugin code which bundles the plugin but then does not build at all - you need to copy over the HTML or CSS as the case might be over to your build folder yourself and the plugin will read it from there. So plugins can be included only if you choose not to build at all. The intermediate option of building and just calling that file seems unavailable and in most cases these bundles are made to be included anyway.

My greater frustration though is that there is no clear/official workflow yet for using Web-Components with ES6 modules (because imports and modules have been developed independently). This become problematic when components dynamically (determined by the server data), I need to use `System.import` with your plugin (and include it externally) instead of <link>  because I'll be using using Javascript to load that component and not HTML. There is of course `Polymer.Base.importHref` but that means I include polymer and use two different systems, so I don't like this solution as well.  

The reason I am writing this is to know what you think about this and if my line of reasoning about the 'build and include' makes sense???

**Comment by Hypercubed** [comment]
I think build and include is a reasonable compromise at this point.  The systemjs-plugin-html plugin does basically the same thing as `Polymer.Base.importHref` which is simulating an HTML import.  After vulcanizing all HTML imports into a single HTML files you either need to use a HTML import (as I have suggested) or use a helper like `Polymer.Base.importHref` or `systemjs-plugin-html`.  This be comes a chicken and egg.

If you have a better idea feel free to comment or reopen.

**Comment by CxRes** [comment]
So I found something that works for my circumstance of dynamic loading `System.import(something.html!)`. I simply import the plugin as `import html` first.

This `import html` was initially not working (which led me to explore these other alternatives) because there was a bug in SystemJS which has been resolved since v0.19.9.

The only suggestion I have is that maybe you can document this trick for those looking to import the build from within or do so dynamically based on some condition.

.
